pkg/export/otel/metrics_internal.go

```diff
@@ -175,3 +186,27 @@ func (p *InternalMetricsReporter) InstrumentationError(processName, errorType st
 		attribute.String("error_type", errorType),
 	))
 }
+
+func (p *InternalMetricsReporter) recordAvoidedService(serviceName, serviceNamespace, serviceInstanceID, serviceVersion, telemetryType string) {
+	attrs := []attribute.KeyValue{
+		semconv.ServiceName(serviceName),
+		semconv.ServiceNamespace(serviceNamespace),
+		semconv.ServiceInstanceID(serviceInstanceID),
+		attribute.String("telemetry_type", telemetryType),
+	}
+	
+	// Only add service version if it's not empty
+	if serviceVersion != "" {
```

Comment:
Are you sure this works? We always set the labels in certain order, if the values don't match, I wonder if we'll end up storing a wrong value for certain label?